Source/Charts/Renderers/BarChartRenderer.swift

```diff
@@ -377,9 +377,9 @@ open class BarChartRenderer: BarLineScatterCandleBubbleRenderer
         // draw the bar shadow before the values
         if dataProvider.isDrawBarShadowEnabled
         {
-            for j in stride(from: 0, to: buffer.rects.count, by: 1)
+            for firstIndexInBar in stride(from: 0, to: buffer.rects.count, by: 1)
```

Comment:
why change name here? seems nothing about `first index`

Reply:
I have the same question. If anything the change should be:
```
 for barRect in buffer.rects
    where viewPortHandler.isInBoundsLeft(barRect.maxX)
{
    if (!viewPortHandler.isInBoundsRight(barRect.origin.x))
    {
          break
    }
    context.setFillColor(dataSet.barShadowColor.cgColor)
    context.fill(barRect)
}
```

Reply:
Seems like it was done for the stacked bar char case, if round each bar when we will have this situation (from my rounding implementation).
![StackedBarChar_rounded](https://user-images.githubusercontent.com/6350992/88600615-5856f180-d077-11ea-8309-95c2a31a17ba.png)
Author's example:
![](https://user-images.githubusercontent.com/5767416/48838216-04671600-ed88-11e8-892f-9524359efeac.png)